### PR TITLE
Codechange: Simplify autoreplace rail/road types by using separate widget.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

The autoreplace window reuses one dropdown list widget index for rail and road type lists. This means the window needs to also check whether it is dealing with trains or road vehicles.

As the widget layout is different for each vehicle type, it is trivial to use a different widget index instead.

## Description

Use a separate widget for rail and road type lists.

This avoids needing to determine which type of list to deal with by additionally checking the window number for VEH_TRAIN/VEH_ROAD.

Does not affect layout so no screenshots.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/autoreplace_gui.cpp [hunk 1]
```diff
@@ -357,28 +352,21 @@ class ReplaceVehicleWindow : public Window {
...
 
-			case WID_RV_RAIL_ROAD_TYPE_DROPDOWN: {
+			case WID_RV_RAIL_TYPE_DROPDOWN: {
 				Dimension d = {0, 0};
-				switch (this->window_number) {
-					case VEH_TRAIN:
-						for (RailType rt = RAILTYPE_BEGIN; rt != RAILTYPE_END; rt++) {
-							const RailTypeInfo *rti = GetRailTypeInfo(rt);
-							/* Skip rail type if it has no label */
-							if (rti->label == 0) continue;
```
Comment (2TallTyler) [inline src/autoreplace_gui.cpp:367]: What happened to this check? Is it no longer needed? (Here and with road types below)

File: src/autoreplace_gui.cpp [hunk 2]
```diff
@@ -662,6 +640,18 @@ class ReplaceVehicleWindow : public Window {
 		}
 	}
 
+	void OnChangeRailRoadType()
```
Comment (2TallTyler) [inline src/autoreplace_gui.cpp]: Missing doxygen

File: src/widgets/autoreplace_widget.h [hunk 3]
```diff
@@ -32,12 +32,13 @@ enum ReplaceVehicleWidgets {
 	WID_RV_INFO_TAB,                 ///< Info tab.
 	WID_RV_STOP_REPLACE,             ///< Stop Replacing button.
 
-	/* Train/road only widgets */
-	WID_RV_RAIL_ROAD_TYPE_DROPDOWN,  ///< Dropdown menu about the rail/roadtype.
-
 	/* Train only widgets. */
+	WID_RV_RAIL_TYPE_DROPDOWN, ///< Dropdown menu about the railtype.
```
Comment (2TallTyler) [inline src/widgets/autoreplace_widget.h]: Since you're here...

`///< Dropdown to select railtype`?

File: src/autoreplace_gui.cpp (hunk 1 above)
Comment (PeterN) [reply src/autoreplace_gui.cpp:367]: `_sorted_railtypes`/`_sorted_roadtypes` already excludes types with no label.

File: src/autoreplace_gui.cpp (hunk 2 above)
Comment (PeterN) [reply src/autoreplace_gui.cpp]: Done, also renamed it to read better, and moved it to private.

**Review by 2TallTyler — APPROVED** [review]

**Pull request merged**